### DISCUSSION TITLE
max_size must be greater than 0 for limit entry 1

### DISCUSSION THREAD

**Discussion by fluca1978** [pgagroal/pgagroal#199] [General]
I'm not sure about the error message I'm getting when trying to start pgagroal: `max_size must be greater than 0 for limit entry 1`.
I suspect the "limit entry 1 " refers to the first line of the `pgagroal_databases.conf` file:

```
$ head -n 2 /etc/pgagroal/pgagroal_databases.conf
# DATABASE USER MAX INITIAL MIN
pgbench pgbench 150 1  1
```

according to this, there is a prefill of one connection. I've set the `max_connections` to 1 in order to do some experiment.
Therefore, I would expect pgagroal to start, prefill a connection and then to respond to other incoming connections that the pool is full.
What am I missing here?

**Comment by jesperpedersen** [comment]
What is your `max_connections` in `pgagroal.conf` ? Or is it only 1 ? If that is the case we can improve the error message as `max_connections` needs to be greater or equal to all `MAX` in `pgagroal_databases.conf`.

**Reply by fluca1978** [reply]
> What is your `max_connections` in `pgagroal.conf` ? Or is it only 1 ? If that is the case we can improve the error message as `max_connections` needs to be greater or equal to all `MAX` in `pgagroal_databases.conf`.

It was set tp 10. sp I need to have `max_connections` greater than any `MAX` or the sum of `MAX`? Because I believe `max_connections` should be at least the sum of `INITIAL`, then I would expect to get a "pool is full" error if I open more connections.

**Reply by jesperpedersen** [reply]
`max_connections` needs to be >= to the sum of all MAX as `pgagroal_databases.conf` only limits users and databases.

Clearly the error message could be better in this case. Feel free to take a look at https://github.com/agroal/pgagroal/blob/master/src/libpgagroal/configuration.c#L1351 to improve this.

Thanks for your feedback !

**Comment by fluca1978** [comment]
I've tried to improve the situation: see #202 .
Unluckily, the idea is that during the parsing of the configuration file a single limit entry is zeroed if no more available connections are available. This means the user will never "see" the real `FATAL` error (that is "too much MAX connection within limits"), rather the fact that an entry has a zero value.

**Comment by jesperpedersen** [comment]
In the validate method once you know there is an error you can gather all the information and present that to the user
